src/realm/object-store/impl/collection_notifier.cpp

```diff
@@ -94,26 +88,47 @@ void DeepChangeChecker::find_related_tables(DeepChangeChecker::RelatedTables& ou
         for (auto& incoming_link : incoming_link_columns) {
             REALM_ASSERT(incoming_link.first);
             auto& links = complete_mapping[incoming_link.first];
-            links.link_columns.insert(incoming_link.second);
-            links.connected_tables.insert(it_table->get_key());
+            links.link_columns.push_back(incoming_link.second);
+            links.connected_tables.push_back(it_table->get_key());
         }
     }
 
-    std::unordered_set<TableKey> tables_to_check = {table.get_key()};
-    std::unordered_set<TableKey> checked_tables;
+    // Remove duplicates:
+    // duplicates in link_columns can occur when a Mixed(TypedLink) contain links to different tables
+    // duplicates in connected_tables can occur when there are different link paths to the same table
+    for (auto& it : complete_mapping) {
+        LinkInfo& it_link = it.second;
+        std::sort(it_link.link_columns.begin(), it_link.link_columns.end());
+        auto last_cols = std::unique(it_link.link_columns.begin(), it_link.link_columns.end());
```

Comment:
Normally I'd say that whether or not to use a temporary variable here is purely taste, but `foo.erase(std::unique(foo.begin(), foo.end()), foo.end());` is one of those standard C++ container/algorithm idioms that is typically written as a single statement. If the problem is that the line is too long, maybe add an alias reference to the container shorten it that way?

(or ignore this entirely because it's the most trivial of nits possible)

Reply:
I went for the reference aliasing. This and adding in the erase/unique combination really cleans up the verbosity here and makes it much nicer to read.